tests/components/shelly/conftest.py

```diff
@@ -183,10 +211,13 @@ async def rpc_wrapper(hass):
 
     hass.data[DOMAIN] = {DATA_CONFIG_ENTRY: {}}
     hass.data[DOMAIN][DATA_CONFIG_ENTRY][config_entry.entry_id] = {}
+    hass.data[DOMAIN][DATA_CONFIG_ENTRY][config_entry.entry_id][
```

Comment:
We shouldn't touch hass.data in tests. Just set up the integration while patching appropriately.

Reply:
This was already done exactly the same for the RPC wrapper 5 lines down.
I just simply added the RPC_POLL wrapper also since this is needed for the update entity.

If the way the tests were already done is wrong, I suggest one of the maintainers of the shelly integration has a look at it since I don't know to much about the shelly integration.

Reply:
I was thinking if I should mention it when approving the PR (that this is known and will be addressed later), 

@MartinHjelmare you already commented this on one of my previous PRs, we even discuss it briefly and I am working on rewriting the tests so we can do a standard init of the integration.

Unfortunately I can't keep up with the changes to the integration so it is delayed, but I do hope to finish it in the next two weeks, mainly because it also blocks https://github.com/home-assistant/core/pull/76980

Reply:
Ok, thanks.